integration/spark2/src/main/scala/org/apache/carbondata/indexserver/DistributedRDDUtils.scala

```diff
@@ -0,0 +1,157 @@
+ * limitations under the License.
+ */
+package org.apache.carbondata.indexserver
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
+
+import org.apache.commons.lang.StringUtils
+import org.apache.hadoop.mapreduce.InputSplit
+import org.apache.spark.Partition
+
+import org.apache.carbondata.core.datamap.Segment
+import org.apache.carbondata.core.datamap.dev.expr.DataMapDistributableWrapper
+
+object DistributedRDDUtils {
+  // Segment number to executorNode mapping
+  val segmentToExecutorMapping: java.util.Map[String, String] =
+    new ConcurrentHashMap[String, String]()
+
+  // executorNode to segmentSize mapping
+  val executorToCacheSizeMapping: java.util.Map[String, Long] =
+    new ConcurrentHashMap[String, Long]()
+
+  def getExecutors(segment: Array[InputSplit], executorsList : Set[String],
+      tableUniqueName: String, rddId: Int): Seq[Partition] = {
+    // sort the partitions in increasing order of index size.
+    val (segments, legacySegments) = segment.span(split => StringUtils
+      .isNotEmpty(split
+      .asInstanceOf[DataMapDistributableWrapper].getDistributable.getSegment.getIndexSize))
+    val sortedPartitions = segments.sortWith(_.asInstanceOf[DataMapDistributableWrapper]
+                                              .getDistributable.getSegment.getIndexSize >
+                                            _.asInstanceOf[DataMapDistributableWrapper]
+                                              .getDistributable.getSegment.getIndexSize)
+    val executorCache = DistributedRDDUtils.executorToCacheSizeMapping
+    // check if any executor is dead.
+    if (executorsList.size < executorCache.size) {
+      // extract the dead executor host name
+      val invalidExecutors = executorCache.keySet().asScala.diff(executorsList)
+      DistributedRDDUtils.invalidateExecutors(invalidExecutors.toSeq)
+    }
+    convertToPartition(legacySegments, tableUniqueName, executorsList, rddId) ++
+    convertToPartition(sortedPartitions, tableUniqueName, executorsList, rddId)
+  }
+
+  private def convertToPartition(segments: Seq[InputSplit], tableUniqueName: String,
+      executorList: Set[String], rddId: Int): Seq[Partition] = {
+    segments.zipWithIndex.map { case (partition, index) =>
+      val wrapper = partition.asInstanceOf[DataMapDistributableWrapper]
+        .getDistributable
+      if (StringUtils.isEmpty(wrapper.getSegment.getIndexSize)) {
+        wrapper.getSegment.setIndexSize("1")
+      }
+      wrapper.setLocations(Array(DistributedRDDUtils
+        .assignExecutor(tableUniqueName, wrapper.getSegment, executorList)))
+      new DataMapRDDPartition(rddId, index, partition)
+    }
+  }
+
+  /**
+   * Update the cache size returned by the executors to the driver mapping.
+   */
+  def updateExecutorCacheSize(cacheSizes: Set[String]): Unit = {
+    synchronized {
+      cacheSizes.foreach {
+        executorCacheSize =>
+          val size = executorCacheSize.split("_")
```

Comment:
For better understanding can you put a dummy value for `executorCacheSize` so that we get a fair idea of how the value of this variable look like and what will be its output after splitting

Reply:
explained in comments